tests/test_frameDict.py

```diff
@@ -0,0 +1,299 @@
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+import astshim as ast
+from astshim.test import MappingTestCase
+
+
+class TestFrameDict(MappingTestCase):
+
+    def setUp(self):
+        self.frame1 = ast.Frame(2, "Domain=frame1")
+        self.frame2 = ast.Frame(2, "Domain=frame2")
+        self.zoom = 1.5
+        self.zoomMap = ast.ZoomMap(2, self.zoom, "Ident=zoomMap")
+        self.initialNumFrames = self.frame1.getNObject()  # may be >2 when run using pytest
+        self.initialNumZoomMap = self.zoomMap.getNObject()  # may be > 1 when run using pytest
+
+    def checkDict(self, frameDict):
+        for index in range(1, frameDict.nFrame + 1):
+            domain = frameDict.getFrame(index).domain
+            self.assertEqual(frameDict.getIndex(domain), index)
+            self.assertEqual(frameDict.getFrame(domain).domain, domain)
+
+    def test_FrameDictOneFrameConstructor(self):
+        frameDict = ast.FrameDict(self.frame1)
+        self.assertIsInstance(frameDict, ast.FrameDict)
+        self.assertEqual(frameDict.nFrame, 1)
+        self.assertEqual(frameDict.getAllDomains(), ["FRAME1"])
+        self.assertEqual(frameDict.getIndex("frame1"), 1)  # should be case blind
+
+        with self.assertRaises(ValueError):
+            frameDict.getIndex("missingDomain")
+        with self.assertRaises(ValueError):
+            frameDict.getIndex("")
+
+        # Make sure the frame is deep copied
+        self.frame1.domain = "NEWDOMAIN"
+        self.assertEqual(frameDict.getFrame("FRAME1").domain, "FRAME1")
+        self.assertEqual(frameDict.getFrame(frameDict.BASE).domain, "FRAME1")
+        self.assertEqual(self.frame1.getRefCount(), 1)
+        self.assertEqual(self.frame1.getNObject(), self.initialNumFrames + 1)
+
+        # make sure BASE and CURRENT are available on the class and instance
+        self.assertEqual(ast.FrameDict.BASE, frameDict.BASE)
+        self.assertEqual(ast.FrameDict.CURRENT, frameDict.CURRENT)
+
+        self.checkCopy(frameDict)
+
+        indata = np.array([
+            [0.0, 0.1, -1.5],
+            [5.1, 0.0, 3.1],
+        ])
+        self.checkMappingPersistence(frameDict, indata)
+        self.checkPersistence(frameDict)
+        self.checkDict(frameDict)
+
+    def testFrameDictFrameSetConstructor(self):
+        frameSet = ast.FrameSet(self.frame1, self.zoomMap, self.frame2)
+        frameDict = ast.FrameDict(frameSet)
+
+        indata = np.array([[1.1, 2.1, 3.1], [1.2, 2.2, 3.2]])
+        predictedOut = indata * self.zoom
+        assert_allclose(frameDict.applyForward(indata), predictedOut)
+        assert_allclose(frameDict.applyInverse(predictedOut), indata)
```

Comment:
I prefer the more direct approach. If this fails it might indicate a bug in `FrameSet` which would be papered over by comparing against `FrameSet.apply...`.